### ISSUE TITLE
Update Debug JVM Parameters From Latest Lambda Settings

### ISSUE THREAD

**Issue by jaimm** [awslabs/aws-sam-cli#524]
https://github.com/awslabs/aws-sam-cli/blob/9ae4b80313d094d3eded4c5ca61731481b36c950/samcli/local/docker/lambda_container.py#L180-L182
Please see the corresponding issue (and commit) with docker-lambda for the latest values:
https://github.com/lambci/docker-lambda/issues/102
The JVM memory parameters are currently out of date with what Lambda uses in production. docker-lambda has updated their settings, but you override them when users pass the debugging flags.

**Comment by jfuss** [comment]
@jaimm Thanks for the issue. Would you be up for send a PR with the updated values?

**Comment by jaimm** [comment]
Sorry, I'm pretty busy right now, otherwise I would. It would be one thing if making this change would help my team, but since Amazon internally uses an old fork of this code, it's hard to justify the time spent, however small. I just wanted to give you guys a heads-up with this issue since I already convinced docker-lambda to make the change.

**Comment by jason-riddle** [comment]
Looks like the values are set here:

https://github.com/lambci/docker-lambda/blob/8eb1919506c9a947bda11770fbbe764ea6f1d9ec/java8/run/Dockerfile#L20-L22

```dockerfile
ENTRYPOINT ["/usr/bin/java", "-XX:MaxHeapSize=2834432k", "-XX:MaxMetaspaceSize=163840k", "-XX:ReservedCodeCacheSize=81920k", \
  "-XX:+UseSerialGC", "-Xshare:on", "-XX:-TieredCompilation", "-Djava.net.preferIPv4Stack=true", \
  "-jar", "/var/runtime/lib/LambdaJavaRTEntry-1.0.jar"]
```

I'll open a PR for this.

**Comment by jfuss** [comment]
Closing this as it was released in [0.6.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.6.0)
